index.html

```diff
@@ -86,25 +86,311 @@ <h3>Use-case #4: Avoiding "Hall of Mirrors"</h3>
+        <p>
+          When a {{MediaStreamTrack}} is a video track derived from screen-capture
+          of a <a data-cite="SCREEN-CAPTURE#dfn-browser">browser</a>
+          <a data-cite="SCREEN-CAPTURE#dfn-display-surface">display surface</a>,
+          {{MediaStreamTrack/getSupportedCaptureActions}} returns the set of
+          available [=capture actions=], if any, supported by the captured
+          application associated with this video track.
+        </p>
+        <pre class="idl">
+          partial interface MediaStreamTrack {
+            sequence&lt;DOMString&gt; getSupportedCaptureActions();
+            Promise&lt;undefined&gt; sendCaptureAction(CaptureAction action);
+          };
+        </pre>
+        <dl data-link-for="MediaStreamTrack" data-dfn-for="MediaStreamTrack">
+          <dt>
+            <dfn>getSupportedCaptureActions</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST return [=this=]'
+            {{MediaDevices/[[AvailableCaptureActions]]}} if defined, or `[]` if not defined.</p>
+          </dd>
+          <dt>
+            <dfn>sendCaptureAction</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST run the following steps:</p>
+            <ol>
+              <li>
+                If the [=relevant global object=] of [=this=] does not have
+                [=transient activation=], return a promise [=rejected=] with
+                {{InvalidStateError}}.
+              </li>
+              <li>
+                [=Consume user activation=].
+              </li>
+              <li>
+                Let |action| be the method's first argument.
+              </li>
+              <li>
+                If |action| is not in [=this=]' {{MediaDevices/[[AvailableCaptureActions]]}},
+                return a promise [=rejected=] with {{NotFoundError}}.
+              </li>
+              <li>
+                Let |p| be a new promise.
+              </li>
+              <li>
+                Run the following steps in parallel:
+                <ol>
+                  <li>
+                    <p>
+                      Queue a task on the task-list of the captured
+                      <a data-cite="SCREEN-CAPTURE#dfn-browser">browser</a>
+                      <a data-cite="SCREEN-CAPTURE#dfn-display-surface">display surface</a>'s
+                      [=top-level browsing context=]'s [=active document=] to run the
+                      following steps:
+                    </p>
+                    <ol>
+                      <li>
+                        Let |target| be the the [=relevant settings object=]'s
+                        [=environment settings object/responsible document=]'s
+                        associated navigator's {{MediaDevices}} object.
+                      <li>
+                        If |action| is not in |target|'s
```

Comment:
IIUC, the real value presented by the Promise is to figure out if there was an asynchronous change in the captured surface's set of supported actions and rejecting then. For anything else, a Promise is not necessary - AFAICT. I'd rather get rid of the Promise then.

Reply:
This wasn't clear from the slides, but since the other tab is in a different process, having a promise that resolves after the action has been delivered sounds useful to me.

For instance, if in the future we want to allow a captureee to update its set of actions in response to an action, then the capturer could deterministically read these new values without polling or needing an event:
```js
const oldActions = track.getSupportedCaptureActions(); // ["next"]
await sendCaptureAction("next");
const newActions = track.getSupportedCaptureActions(); // ["previous", "next"]
if (newActions != oldActions) {
  updateButtons(newActions);
}
```
But even if we never do this, having a promise ACK seems useful.

Reply:
Please see my other comment for why I think it is undesirable to return a Promise that resolves when the other side receives the message.